src-tauri/src/manager.rs

```diff
@@ -175,6 +206,23 @@ fn handle(rx: Receiver<WatcherMessage>, state: Arc<Mutex<ManagerState>>) {
     }
 }
 
+fn wrapper_start_watcher(tx: Sender<WatcherMessage>) {
+    loop {
+        let (lock, cvar) = &*WATCHER_CONDVAR;
+        let mut watchers = lock.lock().unwrap();
+        while !*watchers {
+            watchers = cvar.wait(watchers).unwrap();
+        }
+
+        let mut state = WATCHER_STATE.lock().unwrap();
+        for watcher in state.drain(..) {
+            start_watcher(watcher, tx.clone());
+        }
+        *watchers = false;
+        drop(watchers);
+        drop(state)
+    }
+}
```

Comment:
Waits on a condition variable that alerts it that there's a new watcher in ```WATCHER_STATE```.  It then passes that to start watcher and releases the mutexes

Reply:
Would be good to leave comments like this in the code!

Reply:
Removed this entire function